data/data/baremetal/masters/main.tf

```diff
@@ -1,3 +1,11 @@
+# This null resource is used to wait for the Ironic API to become available before trying to create
+# the master hosts.
+resource "null_resource" "api-wait" {
+  provisioner "local-exec" {
+    command = "timeout 900 bash -c 'while [[ \"$(curl -s -o /dev/null -w \"%%{http_code}\" ${var.ironic_uri})\" != \"200\" ]]; do sleep 5; done'"
```

Comment:
I was hoping to have a different way to do this, I tried some things on Friday, but it's tricky. Terraform needs the provider to do the planning stage, so I can't actually block in the provider.  I could do the waiting for each individual resource but that looked worse than this hack.  

This is a known issue: https://github.com/hashicorp/terraform/issues/2430, it's a common problem that "Provider X is responsible for bringing up the infrastructure that provider Y uses."

Reply:
Ack, I've been testing this locally and it seems to work fine, perhaps we should go with the curl for now, and be ready to rework it in future?

As discussed yesterday we also need a way to wait for the images, which could be similar e.g `curl -I http://127.0.0.1/images/rhcos-ootpa-latest.qcow2.md5sum` and similar for the IPA kernel/ramdisk?